apps/next/src/components/TokenSelect/TokenSelect.tsx

```diff
@@ -34,16 +41,99 @@ function TokenSelectRaw({
   disabled,
 }: TokenSelectProps) {
   const { t } = useTranslation();
+  const { getNetwork } = useNetworkContext();
+  const [selectedChainId, setSelectedChainId] = useState<
+    number | 'avalanche' | null
+  >(null);
+
+  // Helper to check if a chain ID is any Avalanche network (C, X, or P)
+  const isAnyAvalancheNetwork = useCallback(
+    (chainId: number): boolean => {
+      const network = getNetwork(chainId);
+      return (
+        isAvalancheChainId(chainId) ||
+        isXchainNetwork(network) ||
+        isPchainNetwork(network)
```

Comment:
You're using `getNetwork` to check if `chainId` belongs to an Avalanche chain (btw. we have `isPchainNetworkId` and `isXchainNetworkId` functions, no need to `getNetwork` here).

Then you're using `getNetwork` again to pull only the `chainName` property from the result (line `122`).
And then later, in `ChainFilterChips.tsx:83`, you're using `getNetwork()` again to get the logo.

Please choose one place to retrieve all the required network information all at once.